app/main.py

```diff
@@ -173,57 +193,95 @@ def generate_prometheus_metrics(metrics):
+            if namespace == "SYS.ELB" and len(m["dimensions"]) > 1:
+                continue
+
+            metric = {
+                "namespace": namespace,
+                "dimensions": [
+                    {
+                        "name": dimensions_name,
+                        "value": dimensions_value
+                    }
+                ],
+                "metric_name": metric_name
+            }
+
+            metrics_set.append(metric)
+
+            # After each 10 metrics query them as a batch
+            if idx % 10 == 0 or idx == len(filtered):
+
+                # Create payload for batch request
+                payload = {
+                    "metrics": metrics_set,
+                    "from": current_time[0],
+                    "to": current_time[1],
+                    "period": "300",
+                    "filter": "average"
+                }
+
+                # Reset metrics_set for next batch
+                metrics_set = []
+
+
+                # Wait for 100 milliseconds before every request (needed to don't overload OTC API)
+                time.sleep(0.1)
+                r = requests.post(cloud_eye_base, headers={'X-Auth-Token': get_token()},
+                                data=json.dumps(payload))
+
+                if r.status_code == 200:
+                    resp = json.loads(r.text)
+                    if resp['metrics']:
+                        for mr in resp['metrics']:
+                            if len(mr['datapoints']) > 0:
+                                namespace = mr["namespace"]
+                                metric_name = mr["metric_name"]
+                                dimensions_name = mr["dimensions"][0]["name"]
+                                dimensions_value = mr["dimensions"][0]["value"]
+                                full_metric_name = "%s:%s" % (namespace, metric_name)
+                                resource_name = get_resource_name(resource_kind=namespace.replace("SYS.", ""),
+                                                                resource_id=dimensions_value)
+
+                                prometheus_metrics[full_metric_name].labels(unit=mr['unit'],
+                                                                            resource_id=dimensions_value,
+                                                                            resource_name=resource_name).set(mr['datapoints'][0]['average'])
+    
+                                logging.debug("{0} for '{1}={2}' ({3}) at {4} : {5}".format(full_metric_name, dimensions_name,
+                                                                                dimensions_value, resource_name,
+                                                                                datetime.fromtimestamp(mr['datapoints'][0]['timestamp'] / 1000.0),
+                                                                                mr['datapoints'][0]['average']))
+
+
+                elif r.status_code == 401:
```

Comment:
Would have to find out wich return code the API has for throttling and the exponential backoff retry strategy could be added here.

Reply:
According to the docs https://docs.otc.t-systems.com/en-us/api/apiug/apig-en-api-180328012.html it seems that there's no response code specific to throttling.

Reply:
https://docs.otc.t-systems.com/api/ces/ces_03_0023.html mentions an HTTP 429 for too many requests.